### DISCUSSION TITLE
WIGOS Station Identifiers

### DISCUSSION THREAD

**Discussion by craigrj** [dtcenter/METplus#1597] [File I/O] [answered]
We are starting to get observations with the WIGOS identifier - can be up to 20 characters.  I wanted to make sure MET SID code will not have any issue with long station identifier filenames.  Is there a length limit?

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @craigrj, glad this is an easy one! The upper limit for the station ID string length is 40 characters. I tested this out using ascii2nc by appending a lot of characters to the end of an existing station ID (from "72365" to "72365_THIS_IS_A_LONG_SID_THAT_IS_OVER_40_CHARACTERS_LONG").

Running the result through ascii2nc produces this WARNING message:
```
WARNING: 
WARNING:   write_nc_string_array()  truncated a string   from "72365_THIS_IS_A_LONG_SID_THAT_IS_OVER_40_CHARACTERS_LONG" to "72365_THIS_IS_A_LONG_SID_THAT_IS_OVER_40"
WARNING: 
```

There is actually no real upper limit to the station id name in the Point-Stat or Ensemble-Stat tools which processes the point observations. But the point pre-processing tool (ascii2nc, madis2nc, pb2nc, and so on) write the station ID's with maximum character length of 40.

You shouldn't run into any issues with 20 character station names.

**Reply by JohnHalleyGotway** [reply]
@craigrj, FYI, I update the [ascii2nc input format table](https://met.readthedocs.io/en/develop/Users_Guide/reformat_point.html#id8) to indicate the maximum string lengths in the development version of our documentation.

Thanks for pointing out the need for that info!

**Comment by craigrj** [comment]
Thanks for help.

**Reply by JohnHalleyGotway** [reply]
Marking this one as answered and locking discussion on it.
